### PR TITLE
remove host from an old ip if you add again

### LINKED ISSUE #33: Bug: Adding a host can add duplicates with differing IPs
This situation is not handled:

```
when adding hostname with IP
  if ( h.HasHostname( 'example.com' ) AND ! h.Has( ip,'example.com ) ) {
    remove hostname with old IP
```

If I add a hostname `example.com` with `1.1.1.1`, then try to add the same hostname with `2.2.2.2` the result is that I now have 2 entries for `example.com` for both IPs. I expected a single entry for `2.2.2.2`.

When adding a host, old entries should be removed.

This has recently caught us out over at VVV with the vagrant plugin usage when a VirtualBox restriction forced us to change the IP we use. Users who upgraded without halting their VMs kept the hosts file entry, and when they reprovisioned a second entry was added with the new IP. As a result requests to our domains were sent to the old IP, and destroying/halting the VM only triggered the new entry to be removed, the old entry being preserved and persisting the issue.

### PR DESCRIPTION
**Pull request by luthermonson**
closes #33

### CODE REVIEW

File: hosts.go [hunk 1]
```diff
@@ -149,6 +149,19 @@ func (h *Hosts) Add(ip string, hosts ...string) error {
 		return fmt.Errorf("%q is an invalid IP address", ip)
 	}
 
+	// remove hosts from other ips if it already exists
+	for _, host := range hosts {
+		for _, p := range h.getHostPositions(host) {
+			if h.Lines[p].IP == ip {
+				break
```
Comment (tomjn) [inline hosts.go:156]: should this be a continue to next iteration instead? If the first line of the hosts file has the desired IP but the next line does not, then it won't catch it

Comment (luthermonson) [reply hosts.go:156]: ah yes, continue... good catch

Comment (luthermonson) [reply hosts.go]: still passed tests tho... scary :D

**Review by tomjn — APPROVED** [review]
Code review looks good, not been able to run the code locally but approving

**Pull request merged**